include/verilated_imp.h

```diff
@@ -448,36 +447,105 @@ class VerilatedImp {
 
 public:  // But only for verilated*.cpp
     // METHODS - file IO
-    static IData fdNew(FILE* fp) VL_MT_SAFE {
+    static IData fdNewMcd(const char* filenamep) VL_MT_SAFE {
+        VerilatedLockGuard lock(s_s.m_fdMutex);
+        if (s_s.m_fdps.empty()) initialize_fp_table();
+        if (s_s.m_fdFreeMct.empty()) return 0;
+        IData idx = s_s.m_fdFreeMct.back();
+        s_s.m_fdFreeMct.pop_back();
+        s_s.m_fdps[idx] = fopen(filenamep, "w");
+        if(VL_UNLIKELY(!s_s.m_fdps[idx])) return 0;
+        return (1 << idx);
+    }
+    static IData fdNew(const char* filenamep, const char* modep) VL_MT_SAFE {
+        FILE* fp = fopen(filenamep, modep);
         if (VL_UNLIKELY(!fp)) return 0;
+        if (s_s.m_fdps.empty()) initialize_fp_table();
         // Bit 31 indicates it's a descriptor not a MCD
         VerilatedLockGuard lock(s_s.m_fdMutex);
         if (s_s.m_fdFree.empty()) {
             // Need to create more space in m_fdps and m_fdFree
-            size_t start = s_s.m_fdps.size();
-            s_s.m_fdps.resize(start * 2);
-            for (size_t i = start; i < start * 2; ++i) {
-                s_s.m_fdFree.push_back(static_cast<IData>(i));
-            }
+            const size_t start = std::max(31ul + 1ul + 3ul, s_s.m_fdps.size()), excess = 10;
```

Comment:
```suggestion
            const size_t start = std::max(31ul + 1ul + 3ul, s_s.m_fdps.size());
            const size_t excess = 10;
```
(cpp-check style warning otherwise)